command-parser.js

```diff
@@ -209,9 +204,11 @@ var parse = exports.parse = function (message, room, user, connection, levelsDee
 	var commandHandler;
 	do {
 		commandHandler = currentCommands[cmd];
-		if (typeof commandHandler === 'string') {
-			// in case someone messed up, don't loop
+		var failsafe = 0;
+		while (typeof commandHandler === 'string') {
+			if (failsafe >= 10) return connection.sendTo(room, "Error: Too much recursion");
 			commandHandler = currentCommands[commandHandler];
+			failsafe++;
```

Comment:
I had to add this to allow for recursive aliases. I'm not sure if `return connection.sendTo(room, "Error: Too much recursion");` is the right way to error out of an infinity loop.